core/iwasm/libraries/wasi-nn/src/wasi_nn.c

```diff
@@ -565,17 +617,56 @@ wasi_nn_load_by_name(wasm_exec_env_t exec_env, char *name, uint32_t name_len,
         goto fail;
     }
 
-    res = ensure_backend(instance, autodetect, wasi_nn_ctx);
-    if (res != success)
-        goto fail;
+    if (ends_with(nul_terminated_name, TFLITE_MODEL_FILE_EXT)) {
```

Comment:
name here is not necessarily a filename.
see https://github.com/bytecodealliance/wasm-micro-runtime/issues/4331

Reply:
Just to take a chance, as long as this is a string to check here.